### DISCUSSION TITLE
What am I missing with Accordion.Item's "onToggle"?

### DISCUSSION THREAD

**Discussion by mgramigna** [mantinedev/mantine#1649] [Q&A] [answered]
Hello! Thanks for the great component framework :)

I'm trying to invoke a function whenever a specific `Accordion.Item` has been either `expanded` or `collapsed`. I thought of passing in `onToggle` to my accordion items and I saw that it appears to fire `onClick` of the [button that is the header](https://github.com/mantinedev/mantine/blob/b0ac2efb6c04ab34da7b679e825d57acaa9ce735/src/mantine-core/src/components/Accordion/AccordionItem/AccordionItem.tsx#L71) of the `AccordionItem`, which is what I want. However, my `onToggle` just never seems to work. Am i missing something obvious? Is there a different way to achieve what I'm after using `Accordion`? I could envision a workaround using `controlRef` but wasn't sure.

Replication: https://codesandbox.io/s/interesting-frost-3f0czt?file=/src/App.js (note that nothing logs to the console)

**Comment by rtivital** [comment, marked as answer]
Hi Matthew, thanks for kind words! `Accordion.Item` does not support `onToggle` prop, it is not mentioned anywhere in the docs.

**Reply by mgramigna** [reply]
@rtivital Thanks for the quick response! Guess I shouldn't have tried to circumvent the docs and look at the source code ;) 

Then, is there a way to use anything other than an index to identify the different accordion items? Viewing the accordion state would also solve my issue I think, but is the accordion state always going to be `{0: true, 1: false, 2: false, ....}` etc.?

**Reply by rtivital** [reply]
Yes, that's right
